metabase/metabase
# Add SDK storybook PoC

Relates to https://www.notion.so/metabase/Embedding-SDK-e2e-tests-c3c652dd4c5b452a9a243266ae434bb3

### Description

This adds storybook and e2e tests for Embedding SDK.
Storybook acts as an example host app to show Embedding SDK components.
Then we run e2e tests to open storybook story and automate a user flow.

### How to verify

Storybook:
- `yarn dev-ee` to run Metabase.
- Go to `http://localhost:3000/admin/settings/authentication/jwt` and setup JWT secret to be "0000000000000000000000000000000000000000000000000000000000000000"
- So to `http://localhost:3000/admin/settings/embedding-in-other-applications/full-app` and set AUTHORIZED ORIGINS to "`*`"
- `yarn storybook-embedding-sdk`
- Open `http://localhost:6006/?path=/story/embeddingsdk-staticdashboard--default`

e2e tests:
Local:
- Run storybook from the previous step
- Run tests locally with `yarn test-cypress-open-sdk`
- `e2e/test/scenarios/embedding-sdk/static-dashboard.cy.spec.js` scenario should pass

<img width="1150" alt="Screenshot 2024-07-19 at 20 09 55" src="https://github.com/user-attachments/assets/59944fb9-4c28-442c-be23-b9334835816a">


CI:
`e2e-tests-embedding-sdk-ee` CI job should pass

### Demo

<img width="1150" alt="Screenshot 2024-07-13 at 00 06 41" src="https://github.com/user-attachments/assets/543ac714-c6bf-4036-8c61-294ad7891fbf">


### Checklist
